pkg/lightning/backend/external/writer.go

```diff
@@ -57,6 +57,9 @@ var (
 	MergeSortOverlapThreshold int64 = 4000
 	// MergeSortFileCountStep is the step of file count when we split the sorted kv files.
 	MergeSortFileCountStep = 4000
+	// MergeSortMergeFactor assumes each merge sort subtask generates 16 files.
+	// To make sure the files count after merge sort is less than threshold, we should set a lower limit for each subtask.
+	MergeSortMergeFactor = int(MergeSortOverlapThreshold) / 16
```

Comment:
When the subtask concurrency is less than or equal to 16,  each merge sort subtask will generate 16 files as most. But if the subtask concurrency is greater than 16, eg: 32, each merge sort subtask will generate 32 files as most. Maybe we should also consider this situation?

Reply:
what about import-into?

> each merge sort subtask will generate 32 files as most.

we should consider concurrency when calculating the number of files assigned to each subtask